pymc/distributions/timeseries.py

```diff
@@ -33,6 +38,192 @@
 ]
 
 
+class GaussianRandomWalkRV(RandomVariable):
+    """
+    GaussianRandomWalk Random Variable
+    """
+
+    name = "GaussianRandomWalk"
+    ndim_supp = 1
+    ndims_params = [0, 0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("GaussianRandomWalk", "\\operatorname{GaussianRandomWalk}")
+
+    def make_node(self, rng, size, dtype, mu, sigma, init, steps):
```

Comment:
Curious why is this function needed now? Is this check already not covered here?

https://github.com/pymc-devs/pymc/pull/5298/files#diff-5fece42aa9647732b360b6c0e13de089a8408c68ae38b650fba38a1c40322229R102

Reply:
Why this function then? I'm assuming make_node is called when the aesara graph is being made?

In other words why cant this logic be moved to where the value check is? Not saying it should be, just trying to understand :)

Reply:
This check happens immediately when the variable is created, while that other one only when/if the variable is evaluated. We don't want to allow the user to even initialize such variables, whereas we can't do that for the value of the steps, because it might be symbolic/non-static.